src/server/game/AI/PlayerAI/PlayerAI.cpp

```diff
@@ -705,6 +705,29 @@ void PlayerAI::DoAutoAttackIfReady()
         DoMeleeAttackIfReady();
 }
 
+void PlayerAI::CancelAllShapeshifts()
+{
+    std::list<AuraEffect*> const& shapeshiftAuras = me->GetAuraEffectsByType(SPELL_AURA_MOD_SHAPESHIFT);
+    std::set<Aura*> removableShapeshifts;
+    for (AuraEffect* auraEff : shapeshiftAuras)
+    {
+        Aura* aura = auraEff->GetBase();
+        if (!aura)
+            continue;
+        SpellInfo const* auraInfo = aura->GetSpellInfo();
+        if (!auraInfo)
+            continue;
+        if (auraInfo->HasAttribute(SPELL_ATTR0_CANT_CANCEL))
```

Comment:
alright, this code doesn't seem valid for 6.x either, in 6.x there are a few spells with SPELL_ATTR0_CANT_CANCEL, but only usable by monks and warriors. Still noone Passive or Positive